lib/ansible/module_utils/network/dcnm/dcnm.py

```diff
@@ -16,8 +16,60 @@
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 #
 
+import socket
+from ansible.module_utils.common import validation
 from ansible.module_utils.connection import Connection
 
+def validate_list_of_dicts(param_list, spec):
+    """ Validate/Normalize playbook params. Will raise when invalid parameters found.
+    param_list: a playbook parameter list of dicts
+    spec: an argument spec dict
+          e.g. spec = dict(ip=dict(required=True, type='ipv4'),
+                           foo=dict(type='str', default='bar'))
+
+    return: list of normalized input data
+    """
+    v = validation
+    normalized = []
+    invalid_params = []
+    for list_entry in param_list:
+        valid_params_dict = {}
+        for param in spec:
+            item = list_entry.get(param)
+            if item is None:
+                if spec[param].get('required'):
+                    invalid_params.append('{} : Required parameter not found'.format(item))
+                else:
+                    item = spec[param].get('default')
+            else:
+                type = spec[param].get('type')
+                if type == 'str':
+                    item = v.check_type_str(item)
+                elif type == 'int':
+                    item = v.check_type_int(item)
+                elif type == 'bool':
+                    item = v.check_type_bool(item)
+                elif type == 'list':
+                    item = v.check_type_list(item)
+                elif type == 'dict':
+                    item = v.check_type_dict(item)
+                elif type == 'ipv4':
+                    address = item.split('/')[0]
+                    try:
+                        socket.inet_aton(address)
+                    except socket.error:
+                        invalid_params.append('{} : Invalid IPv4 address syntax'.format(item))
+                    if address.count('.') != 3:
+                        invalid_params.append('{} : Invalid IPv4 address syntax'.format(item))
+            valid_params_dict[param] = item
+        normalized.append(valid_params_dict)
+
+    if invalid_params:
+        msg = 'Invalid parameters in playbook: {}'.format('\n'.join(invalid_params))
+        raise Exception(msg)
```

Comment:
It might make sense to just return the invalid_params to the caller and let the caller decided whether to raise. My reason is that I validate my nets first and then my attachments after that (they're nested lists) and I could collect all of the invalid info and only raise after everything is done with validation.

Reply:
I think raising here is fine.  No reason to continue if the input contains invalid data.

Reply:
The issue is that I haven't validated all of the input yet, since I make separate calls for net and att data. So it's probably better to collect all of the errors and raise from the caller instead.

Reply:
I see what you mean.  The only drawback is that we might get inconsistencies in how the errors are logged but I am ok either way.